### PR TITLE
Fs scan forwards compatibility

### LINKED ISSUE #3518: Forwards compatiblity cylc scan issue.
When executing cylc scan in cylc 7 with workflows running in cylc 8, an error occurs, as discussed in issue https://github.com/cylc/cylc-flow/issues/3516.

Make cylc scan skip workflows running in newer versions of cylc: list these suites in --debug mode.

### PR DESCRIPTION
**Pull request by datamel**
These changes partially address #3516
These changes close #3518
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] Does not need tests (why? manually tested for now. No current unit tests for port_scan.py - possibly worth adding some?).

### CODE REVIEW

**Comment by hjoliver** [comment]
The failed tests pass locally for me on this branch.

**Comment by hjoliver** [comment]
And, can't see how this change could cause the failures anyway.  I'll kick travis just in case...

**Comment by datamel** [comment]
> The failed tests pass locally for me on this branch.

Yes, bit strange. Passing locally and on passing Travis on my fork - just pulling to master breaks it. I feel like I'm missing something obvious! I will continue digging today.

**Comment by hjoliver** [comment]
> > The failed tests pass locally for me on this branch.
> 
> Yes, bit strange. Passing locally and on passing Travis on my fork - just pulling to master breaks it. I feel like I'm missing something obvious! I will continue digging today.

It could be Travis's fault...

**Comment by kinow** [comment]
Tried a couple of the failed tests locally and they passed. Looking at Travis logs:

```
(...)
2020-03-18T23:56:22Z DEBUG - Reading file /tmp/tmp.DI7LeYxNez/etc/global.rc
CPU time limit exceeded (core dumped)
2020-03-18T23:56:23Z CRITICAL - failed/XCPU
```

Maybe it was under heavy user load and failed? Cleared the caches, and re-triggered one of the failed builds. :crossed_fingers:

**Comment by hjoliver** [comment]
Good spotting @kinow , I missed the CPU time limit message

**Comment by hjoliver** [comment]
I just checked out this branch, tweaked Travis config to run only the failed tests, and pushed back to my fork ... and they all passed.

So I think we can merge this PR.

**Comment by hjoliver** [comment]
(I see @datamel said on the 7.9.x companion PR that the tests pass on her fork too).

**Comment by hjoliver** [comment]
@kinow, I just looked at the raw Travis logs - it looks like the core dump happens only in a flaky test that passes on 2nd attempt, not in the ones that stay failed.

**Comment by oliver-sanders** [comment]
@datamel what's the status of this, can we revert the debug stuff and try to get this through the test battery?

**Comment by datamel** [comment]
> @datamel what's the status of this, can we revert the debug stuff and try to get this through the test battery?

It is now passing travis test battery. I believe the tests were failing under stress but when they were run indivually they pass.

File: .travis/cover.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 def main():
     # Run tests with virtual frame buffer for X support.
-    if call('xvfb-run -a cylc test-battery --chunk $CHUNK --state=save -j 5',
+    if call('xvfb-run -a cylc test-battery --chunk $CHUNK --state=save -j 5 tests/authentication tests/cylc-scan/03-monitor.t',
```
Comment (oliver-sanders) [inline .travis/cover.py]: Need to revert this change

Comment (datamel) [reply .travis/cover.py]: @oliver-sanders, no probs, I'll sort that out now.

File: lib/cylc/network/port_scan.py [hunk 2]
```diff
@@ -308,6 +309,13 @@ def get_scan_items_from_fs(owner_pattern=None, updater=None):
             reg = os.path.relpath(dirpath, run_d)
             try:
                 contact_data = srv_files_mgr.load_contact_file(reg, owner)
+                cylc_ver = contact_data[srv_files_mgr.KEY_VERSION]
+                major_ver = int(cylc_ver.split(".", 1)[0])
+                if (major_ver > 7):
+                    LOG.debug("Suite %s is running in Cylc version %s "
+                              "and will not be displayed." % (reg, cylc_ver)
+                              )
+                    continue
```
Comment (oliver-sanders) [inline lib/cylc/network/port_scan.py:318]: This aught to be in the `else` branch of this `try`.
